// ng2-markdown-to-html/dist/lib/markdown-to-html.module.js
import { NgModule } from '@angular/core';
import { HttpModule } from '@angular/http';
import { LanguagePipe } from './language.pipe';
import { MarkdownToHtmlComponent } from './markdown-to-html.component';
import { MarkdownToHtmlService } from './markdown-to-html.service';
export var MarkdownToHtmlModule = (function () {
    function MarkdownToHtmlModule() {
    }
    MarkdownToHtmlModule.forRoot = function () {
        return {
            ngModule: MarkdownToHtmlModule,
            providers: [MarkdownToHtmlService],
        };
    };
    MarkdownToHtmlModule.forChild = function () {
        return {
            ngModule: MarkdownToHtmlModule,
        };
    };
    MarkdownToHtmlModule.decorators = [
        { type: NgModule, args: [{
                    exports: [
                        MarkdownToHtmlComponent,
                        LanguagePipe,
                    ],
                    imports: [HttpModule],
                    declarations: [
                        MarkdownToHtmlComponent,
                        LanguagePipe,
                    ],
                },] },
    ];
    /** @nocollapse */
    MarkdownToHtmlModule.ctorParameters = function () { return []; };
    return MarkdownToHtmlModule;
}());
